tests/unit/s2n_early_data_test.c

```diff
@@ -559,21 +567,29 @@ int main(int argc, char **argv)
             EXPECT_NOT_NULL(conn);
             EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "default_tls13"));
             EXPECT_OK(s2n_append_test_chosen_psk_with_early_data(conn, nonzero_max_early_data, &s2n_tls13_aes_256_gcm_sha384));
-            conn->actual_protocol_version = S2N_TLS13;
             conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
 
+            /* Early data not enabled */
+            conn->early_data_state = S2N_EARLY_DATA_REQUESTED;
+            conn->actual_protocol_version = S2N_TLS13;
+            EXPECT_OK(s2n_early_data_accept_or_reject(conn));
+            EXPECT_EQUAL(conn->early_data_state, S2N_EARLY_DATA_REQUESTED);
+
```

Comment:
Shouldn't the state here be EARLY_DATA_NOT_REQUESTED?

Reply:
You can't move from the S2N_EARLY_DATA_REQUESTED to the S2N_EARLY_DATA_NOT_REQUESTED state: that would be rewriting history! We've sent the request, now we have to live with that :)